### ISSUE TITLE
Lookup a child and parents relationship ID when given the child ID and mother or father ID

### ISSUE THREAD

**Issue by justincy** [fs-webcomponents/fs-add-person#5]
This use case comes from the pedigree where we have person IDs but not relationship IDs. Currently the sample app is catching the add events from the pedigree and looking up the child and parents ID when necessary. We should port that functionality into this element.

See https://github.com/fs-webcomponents/sample-app/issues/22

There are a number of use cases for adding persons and wanting certain relationships to be constructed (as [documented](https://www.webcomponents.org/element/fs-webcomponents/fs-add-person/elements/fs-add-person)). We differentiate between the use cases by examining the data that's provided.

In this use case we either have just a childId (adding the first parent to a child) or we have a childId with a fatherId or motherId (adding a second parent to the child). The former use case is already covered.

    <fs-add-person child-id="STGH-123"></fs-add-person>

We can cover the second use case by allowing for this:

    <fs-add-person child-id="STGH-123" mother-id="MMMM-123"></fs-add-person>

We'll have to update the first case to skip creating a new capr if a motherId or fatherId is provided. But there's already a use case that expects a motherId or fatherId without a childId so we'll have to update that one to skip if there's a childId present.

* [x] Increase readability of `_createRelationship()`
* [x] Better use case docs; assign names to the use cases
* [x] Update the new-parent-create-capr use case to skip if a motherId or fatherId is also present.
* [x] Update the new-child-create-capr use case to skip if a childId is also present.
* [x] Accept childId and motherId or fatherId to update an existing capr without knowing the caprId.

**Comment by justincy** [comment]
The code that creates the relationships, after the person is created or selected, assumes that we can create all requests before we start executing any requests. But this use case is going to break that assumption: we have to make one request to get data before we can construct the next request to update the child-and-parents relationship.

This means we'll need to refactor the most complicated piece of code.

**Comment by justincy** [comment]
I've done a little refactoring that significantly increases readability and will make it easier to make further modifications. However in looking at how the code is full of `if` statements and no `if else` statements, I remember that there was a specific reason that we wanted to allow for multiple use cases to be met at the same time. Yet I can't remember why that is. If we make the two adjustments to existing use cases, as documented above, then we may preclude any possibility of meeting multiple use cases at once.

**Comment by justincy** [comment]
I just compared the use cases described in the docs to the use cases allowed in this element as well as the use cases that the sample app uses. They're all distinct. There's no situation that hits multiple scenarios. Perhaps there was in the past before we added allowed a couple relationship to be created with caprIds and intentionally prevented that from trigger the add-child use case; see https://github.com/fs-webcomponents/fs-add-person/commit/69cd8c255280c3a12dabe87f483a6dcb908a1725.

I guess we're past that hurdle.

One more thing I'd like to do is assign names to each use case. I'm frequently getting them confused in my mind. It's hard to look at the use case examples and know exactly what's expected to happen without reading the explanation. I'd like to at least have a short phrase for each use case that's easy to remember.

**Comment by justincy** [comment]
I've done even more refactoring of `_createRelationship()`. Now we have a method that just generates a list of requests and executes them serially. Now I need to add a use case where we must issue one request and process the return data before being able to create additional requests.

I just realized that our use case fits an existing use case after the first request is finished. The first request is issued to calculate the list of caprIds that match the childId + fatherId + motherId combination. Once we have the list of caprIds then our use case fits the existing add-parent use cases. So when that first request returns and we have the list of caprIds, we can clear the childId property and set the caprIds property then run through the code we have now.
